### PR TITLE
Add documentation on datatable options

### LINKED ISSUE #2586: API Documentation checklist
List of symbols exported from the main `datatable` namespace:

## Classes
- [x] `Frame`
    - [x] `__copy__`, `__deepcopy__`
    - [x] `__getitem__`, 
    - [x] `__setitem__`, 
    - [x] `__delitem__`
    - [x] `__getstate__`, `__setstate__`
    - [x] `__init__`
    - [x] `__iter__`
    - [x] `__len__`
    - [x] `__repr__`, `_repr_html_`, `_repr_pretty_`
    - [x] `__reversed__`
    - [x] `__sizeof__`
    - [x] `__str__`
    - [x] `cbind`
    - [x] `colindex`
    - [x] `copy`
    - ~`countna`~, ~`countna1`~
    - [x] `export_names`
    - [x] `head`
    - [x] `key`
    - [x] `keys`
    - [x] `ltypes`
    - [x] `materialize`
    - ~`max`~, ~`max1`~
    - ~`mean`~, ~`mean1`~
    - ~`min`~, ~`min1`~
    - ~`mode`~, ~`mode1`~
    - [x] `names`
    - [x] `ncols`
    - ~`nmodal`~, ~`nmodal1`~
    - [x] `nrows`
    - ~`nunique`~, ~`nunique1`~
    - [x] `rbind`
    - [x] `replace`
    - ~`sd`~, ~`sd1`~
    - [x] `shape`
    - [x] `sort`
    - [x] `source`
    - [x] `stype`
    - [x] `stypes`
    - ~`sum`~, ~`sum1`~
    - [x] `tail`
    - [x] `to_csv`
    - [x] `to_dict`
    - [x] `to_jay`
    - [x] `to_list`
    - [x] `to_numpy`
    - [x] `to_pandas`
    - [x] `to_tuples`
    - [x] `view`
- [x] `FExpr`
    - [x] `__add__`
    - [x] `__and__`
    - [x] `__bool__`
    - [x] `__eq__`
    - [x] `__floordiv__`
    - [x] `__ge__`
    - [x] `__gt__`
    - [x] `__init__`
    - [x] `__invert__`
    - [x] `__le__`
    - [x] `__lshift__`
    - [x] `__lt__`
    - [x] `__mod__`
    - [x] `__mul__`
    - [x] `__ne__`
    - [x] `__neg__`
    - [x] `__or__`
    - [x] `__pos__`
    - [x] `__pow__`
    - [x] `__radd__`
    - [x] `__rand__`
    - [x] `__repr__`
    - [x] `__rfloordiv__`
    - [x] `__rlshift__`
    - [x] `__rmod__`
    - [x] `__rmul__`
    - [x] `__ror__`
    - [x] `__rpow__`
    - [x] `__rrshift__`
    - [x] `__rshift__`
    - [x] `__rsub__`
    - [x] `__rtruediv__`
    - [x] `__rxor__`
    - [x] `__str__`
    - [x] `__sub__`
    - [x] `__truediv__`
    - [x] `__xor__`
    - [x] `extend`
    - [x] `len`
    - [x] `re_match`
    - [x] `remove`
- [x] `stype`
- [x] `ltype`
- [x] `Namespace`
    - [x] `__getitem__`
    - [x] `__getattr__`

## Functions
- [x] `mean` 
- [x] `sd` 
- [x] `min` 
- [x] `max` 
- ~`isna`~
- [x] `sum` 
- [x] `count` 
- [x] `first` 
- ~`abs`~
- ~`exp`~
- [x] `last` 
- ~`log`~
- ~`log10`~
- [x] `median` 
- [x] `cov` 
- [x] `corr`
- [x] `by`
- [x] `cbind`
- [x] `cut`
- [x] `fread`
- [x] `ifelse`
- [x] `init_styles`
- [x] `intersect`
- [x] `iread`
- [x] `join`
- [x] `qcut`
- [x] `rbind`
- [x] `repeat`
- [x] `rowall`
- [x] `rowany`
- [x] `rowcount`
- [x] `rowfirst`
- [x] `rowlast`
- [x] `rowmax`
- [x] `rowmean`
- [x] `rowmin`
- [x] `rowsd`
- [x] `rowsum`
- [x] `setdiff`
- [x] `shift`
- [x] `sort`
- [x] `symdiff`
- [x] `union`
- [x] `unique`
- [x] `update`
- [x] `split_into_nhot`
- ~`open`~

## math.
- [x] `math.`
    - [x] `abs`
    - [x] `arccos`
    - [x] `arcosh`
    - [x] `arcsin`
    - [x] `arctan`
    - [x] `arsinh`
    - [x] `artanh`
    - [x] `atan2`
    - [x] `cbrt`
    - [x] `ceil`
    - [x] `copysign`
    - [x] `cos`
    - [x] `cosh`
    - [x] `deg2rad`
    - [x] `e`
    - [x] `erf`
    - [x] `erfc`
    - [x] `exp`
    - [x] `exp2`
    - [x] `expm1`
    - [x] `fabs`
    - [x] `floor`
    - [x] `fmod`
    - [x] `gamma`
    - [x] `golden`
    - [x] `hypot`
    - [x] `inf`
    - [x] `isclose`
    - [x] `isfinite`
    - [x] `isinf`
    - [x] `isna`
    - [x] `ldexp`
    - [x] `lgamma`
    - [x] `log`
    - [x] `log10`
    - [x] `log1p`
    - [x] `log2`
    - [x] `logaddexp`
    - [x] `logaddexp2`
    - [x] `nan`
    - [x] `pi`
    - [x] `pow`
    - [x] `rad2deg`
    - [x] `rint`
    - [x] `round`
    - [x] `sign`
    - [x] `signbit`
    - [x] `sin`
    - [x] `sinh`
    - [x] `sqrt`
    - [x] `square`
    - [x] `tan`
    - [x] `tanh`
    - [x] `tau`
    - [x] `trunc`

## internal.
- [x] `internal.`
    - [x] `compiler_version`
    - [x] `frame_column_data_r`
    - [x] `frame_columns_virtual`
    - [x] `frame_integrity_check`
    - [x] `get_thread_ids`
    - [x] `in_debug_mode`
    - [x] `regex_supported`

## options
- [x] `options.`
   - [x] `nthreads`
   - [x]  `debug.`
        - [x] `arg_max_size`
        - [x] `enabled`
        - [x] `logger`
        - [x] `report_args`
    - [x] `display.`
        - [x] `allow_unicode`
        - [x] `head_nrows`
        - [x] `interactive` 
        - [x] `max_column_width` 
        - [x] `max_nrows` 
        - [x] `tail_nrows` 
        - [x] `use_colors` 
    - [x] `frame.`
        - [x] `names_auto_index`
        - [x] `names_auto_prefix`
    - [x] `fread.`
        - ~`anonymize`~
        - [x]  `log`.
            - [x] `anonymize` 
            - [x] `escape_unicode`
    - [x] `progress.`
        - [x] `allow_interruption`
        - [x] `callback`
        - [x] `clear_on_success`
        - [x] `enabled`
        - [x] `min_duration`
        - [x] `updates_per_second`
    - [x] `sort.`
        - [x] `insert_method_threshold`
        - [x] `max_chunk_length`
        - [x] `max_radix_bits`
        - [x] `nthreads`
        - [x] `over_radix_bits` 
        - [x] `thread_multiplier` 

## models.
- [x] `Ftrl`
    - [x] `alpha`
    - [x] `beta`
    - [x] `colname_hashes`
    - [x] `colnames`
    - [x] `double_precision`
    - [x] `feature_importances`
    - [x] `fit`
    - [x] `interactions`
    - [x] `labels`
    - [x] `lambda1`
    - [x] `lambda2`
    - [x] `mantissa_nbits`
    - [x] `model`
    - [x] `model_type`
    - [x] `model_type_trained`
    - [x] `nbins`
    - [x] `negative_class`
    - [x] `nepochs`
    - [x] `params`
    - [x] `predict`
    - [x] `reset`
- [x] `aggregate`
- [x] `kfold`
- [x] `kfold_random`

## Other
- [x] `build_info`
- [x] `f `
- [x] `g `
- [x] `bool8`
- [x] `int8`
- [x] `int16`
- [x] `int32`
- [x] `int64`
- [x] `float32`
- [x] `float64`
- [x] `str32`
- [x] `str64`
- [x] `obj64`
- [x] `dt`

### PR DESCRIPTION
**Pull request by oleksiyskononenko**
Closes #2586

### CODE REVIEW

File: docs/api/options/frame.rst [hunk 1]
```diff
@@ -0,0 +1,23 @@
+
+.. xdata:: datatable.options.frame
+    :src: --
+
+    This namespace contains the following :class:`Frame()` options:
```
Comment (st-pasha) [inline docs/api/options/frame.rst:5]: Class shouldn't have the parentheses.

File: src/core/csv/reader.cc [hunk 2]
```diff
@@ -46,6 +46,25 @@ namespace read {
 // options
 //------------------------------------------------------------------------------
 
+static const char * doc_options_fread_log_anonymize =
+R"(
+If `True`, any snippets of data being read that are printed in the
+log will be first anonymized by converting all non-0 digits to `1`,
+all lowercase letters to `a`, all uppercase letters to `A`, and all
+unicode characters to `U`.
+This option is useful in production systems when reading sensitive
```
Comment (st-pasha) [inline src/core/csv/reader.cc:55]: Missing a line break between paragraphs.

File: src/core/frame/names.cc [hunk 3]
```diff
@@ -351,6 +351,24 @@ void Frame::_init_names(XTypeMaker& xt) {
 // Options
 //------------------------------------------------------------------------------
 
+static const char * doc_options_frame_names_auto_index =
+R"(
+When Frame needs to auto-name columns, they will be assigned
+names `C0`, `C1`, `C2`, etc. by default. This option allows you to
+control the starting index in this sequence. For example, setting
+`dt.options.frame.names_auto_index=1` will cause the columns to be
+named `C1`, `C2`, `C3`, etc.
```
Comment (st-pasha) [inline src/core/frame/names.cc:360]: I would add a see-also link to `manual/name_mangling`: 
```
:ref:`name-mangling`
```

**Review by st-pasha — APPROVED** [review]

**Comment by st-pasha** [comment]
I noticed that we don't say in the documentation what is the default value of each option, but this could be handled in another PR if you want.

File: src/core/frame/names.cc (hunk 3 above)
Comment (oleksiyskononenko) [reply src/core/frame/names.cc:360]: Added this section for both frame options.

**Comment by oleksiyskononenko** [comment]
Yeah, we need to think how to better present this information.

**Pull request merged**